### DISCUSSION TITLE
Improved self-authorization workflow for private developers

### DISCUSSION THREAD

**Discussion by jenilew** [amzn/selling-partner-api-models#3111] [Announcements]
Hi All,

We are excited to announce improvements to the self-authorization workflow for private developers!

Private developers can now complete self-authorization across their regional accounts and other seller accounts they own, from the same page instead of switching between accounts from Seller Central. 

As part of the experience, developers can authorize their apps across their regional accounts and retrieve a self-authorization (refresh) token from the same authorization page. 

This change is intended to simplify managing your self-authorization (refresh) tokens across acounts and across regions. 

In Seller Central, go to **Apps & Services**, then **Develop apps**. Click the down arrow next to **Edit Apps** and click **Authorize** to view the new one-stop authorization page. 

### Which marketplaces are affected?
All marketplaces are affected. 

### Who is affected?
Private developers with more than one regional account or more than one seller account are affected. 

### What action is required?
No action is required. 

Best regards,

The Selling Partner API team

**Comment by jenny-AZ** [comment]
I know this is old, but I am hoping someone can help. I have a Private SP-API app and I am trying to authorize it to a different Seller Central account (i.e., not the account that holds the developer profile). It has always worked for me in the past, but now after I sign in to the second account, the Authorize button is simply not there. Did something break? Is there a workaround? I am pretty desperate at this point.

**Reply by ritikacd** [reply]
Hi @jenny-AZ 

From your primary Seller Central account, after you click on `Authorize` button do you see _To authorize an application to access a different account, sign in to that account_?
<img width="1360" alt="Screenshot 2024-11-22 at 5 44 31 PM" src="https://github.com/user-attachments/assets/cc55b36c-dfc7-4cf7-9095-efa953dede2e">

Please note that the private applications have a 10 self-authorizations limit. You can check current self authorizations for your private application on the Manage Authorizations page. This could be the reason that you can no longer self authorize new account.

Thanks,
Ritika
Selling Partner API services.
